backend/src/main.py

```diff
@@ -277,6 +295,93 @@ def friendship_remove():
         return "", status.OK
 
 
+@app.post("/buddy/create")
+def buddy_create():
+    data = flask.request.json
+
+    try:
+        data = dacite.from_dict(data_class=CreateBuddyRequest, data=data)
+    except dacite.DaciteError:
+        return "Wrong fields or data types", status.BAD_REQUEST
+
+    with sqlalchemy.orm.Session(engine) as session:
+        sender_smaller = True
+
+        if data.sender_id > data.receiver_id:
+            sender_smaller = False
+
+        smaller_id, bigger_id = data.sender_id, data.receiver_id
+        if smaller_id > bigger_id:
+            smaller_id, bigger_id = bigger_id, smaller_id
+        friendship = session.get(models.Friendship, (smaller_id, bigger_id))
+
+        if friendship is None or friendship.pending is True:
+            return "Friends before buddies", status.NOT_FOUND
+
+        friendship.smaller_user_id = data.sender_id if sender_smaller else data.receiver_id
+        friendship.bigger_user_id = data.receiver_id if sender_smaller else data.sender_id
+        friendship.sender_is_smaller_id = sender_smaller
```

Comment:
I think you don't need to update these, right? Because the `Friendship` was already in the db, so `smaller_user_id`, `bigger_user_id`, and `sender_is_smaller_id` should've already been set.